pkg/mailer/mailer_test.go

```
@@ -1,35 +1,320 @@
+			_, body = setAuthorityFailureContent(contentData, smtpServer.From, []string{smtpServer.To}, subject)
+			bodyString := body.String()
+			if !strings.Contains(bodyString, contentData.CommonData.Authority) {
+				t.Errorf("Email template %v.html failed. Template malformed. Expected \"%v\" in template, found \"%v\"\n", subject, contentData.CommonData.Authority, "")
+			}
+			if !strings.Contains(bodyString, contentData.CommonData.Username) {
+				t.Errorf("Email template %v.html failed. Template malformed. Expected \"%v\" in template, found \"%v\"\n", subject, contentData.CommonData.Username, "")
+			}
+			if !strings.Contains(bodyString, contentData.CommonData.Name) {
+				t.Errorf("Email template %v.html failed. Template malformed. Expected \"%v\" in template, found \"%v\"\n", subject, contentData.CommonData.Name, "")
+			}
+		case "user-validation-failure-name", "user-validation-failure-email", "user-email-verification-malfunction", "user-creation-failure", "user-serviceaccount-failure",
+			"user-kubeconfig-failure", "user-email-verification-dubious", "user-email-verification-update-malfunction", "user-deactivation-failure":
+			_, body = setUserFailureContent(contentData, smtpServer.From, []string{smtpServer.To}, subject)
+			bodyString := body.String()
+			if !strings.Contains(bodyString, contentData.CommonData.Authority) {
+				t.Errorf("Email template %v.html failed. Template malformed. Expected \"%v\" in template, found \"%v\"\n", subject, contentData.CommonData.Authority, "")
+			}
+			if !strings.Contains(bodyString, contentData.CommonData.Username) {
+				t.Errorf("Email template %v.html failed. Template malformed. Expected \"%v\" in template, found \"%v\"\n", subject, contentData.CommonData.Username, "")
+			}
+			if !strings.Contains(bodyString, contentData.CommonData.Name) {
+				t.Errorf("Email template %v.html failed. Template malformed. Expected \"%v\" in template, found \"%v\"\n", subject, contentData.CommonData.Name, "")
+			}
+
+		}
```

You may also include the Send function of the mailer package.

---

But the send function does not return anything, is there a way to verify that email was really sent?

---

You can return an error at the end of the Send function to do that.

---

I am now executing send() inside a Run test